app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -179,7 +181,14 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         navButtonForward.isFocusable = canGoForward
         navButtonForward.alpha = if (canGoForward) NAVIGATION_BUTTON_ENABLED_ALPHA else NAVIGATION_BUTTON_DISABLED_ALPHA
 
+        val isPinEnabled = navigationStateProvider?.isPinEnabled() ?: false
+        pinButton.isEnabled = isPinEnabled
         pinButton.isChecked = navigationStateProvider?.isURLPinned() ?: false
+        pinButton.alpha = if (isPinEnabled) NAVIGATION_BUTTON_ENABLED_ALPHA else NAVIGATION_BUTTON_DISABLED_ALPHA
```

Comment:
nit: Do we need `isFocusable` too, like the other buttons? Another reason we want helper functions. :)